inc/class-module.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * The Module object provides a common interface for registering
+ * modules and carrying out module related tasks.
+ *
+ * @package hm-platform
+ */
+
+namespace HM\Platform;
+
+class Module {
+	/**
+	 * Store of all registered modules.
+	 *
+	 * @var array
+	 */
+	protected static $modules = [];
+
+	/**
+	 * The slug for the module, used to reference individual modules.
+	 *
+	 * @var string
+	 */
+	protected $slug;
+
+	/**
+	 * The directory where the module lives.
+	 *
+	 * @var string
+	 */
+	protected $directory;
+
+	/**
+	 * The human readable title of the module.
+	 *
+	 * @var string
+	 */
+	protected $title;
+
+	/**
+	 * Settings array.
+	 *
+	 * @var array
+	 */
+	protected $settings;
```

Comment:
Need to document exactly what "settings" means here; it's essentially configuration for the module _itself_